services/ml/execution_impl_android.cc

```diff
@@ -60,19 +62,19 @@ ExecutionImplAndroid::ExecutionImplAndroid(
     result = ANeuralNetworksExecution_setOutput(
         nn_execution_, i, NULL, static_cast<void*>(info->mapping.get()),
         info->length);
-
     DLOG(INFO) << "ANeuralNetworksExecution_setOutput: " << result;
   }
 }
 
 ExecutionImplAndroid::~ExecutionImplAndroid() {
-  ANeuralNetworksExecution_free(nn_execution_);
-  DLOG(INFO) << "ANeuralNetworksExecution_free";
+  // ANeuralNetworksExecution_free(nn_execution_);
```

Comment:
Move `ANeuralNetworksCompilation_free(nn_compilation_ );` from  CompilationImplAndroid to here.